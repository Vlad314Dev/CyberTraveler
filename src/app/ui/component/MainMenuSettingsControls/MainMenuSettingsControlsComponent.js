import './MainMenuSettingsControlsStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
    BACK_OPTION,
    optionLabels,
    SETTINGS_OPTION
} from 'UIStore/MainMenu/MainMenuConfig';

class MainMenuSettingsControlsComponent extends PureComponent
{
    static propTypes = {
        setActiveOption: PropTypes.func.isRequired,
        type: PropTypes.string.isRequired
    };

    setActiveOption = (optionName) => () => {
        const { setActiveOption } = this.props;
        setActiveOption(optionName)
    }

    renderControls()
    {
        const { type } = this.props;

        return (
            <div block="SettingsControls" elem="Wrapper">
                <div block="SettingsControls" elem="Image" mods={ { type: type } }></div>
            </div>
        );
    }

    render()
    {
        return (
            <>
                { this.renderControls() }
                <div block="MainMenuOption" elem="List">
                    {optionLabels.map((option, index) => {
                        if (option.key === BACK_OPTION) {
                            return <div 
                                block="MainMenuOption" 
                                elem="Wrapper" 
                                key={ index }
                            >
                                <span 
                                    block="MainMenuOption" 
                                    elem="Value"
                                    key={ index }
                                    onClick={ this.setActiveOption(SETTINGS_OPTION) }
                                >{ option.label }</span>
                            </div>
                        }
                        
                        return null;
                    })}
                </div>
            </>
        );
    }
}

export default MainMenuSettingsControlsComponent;
