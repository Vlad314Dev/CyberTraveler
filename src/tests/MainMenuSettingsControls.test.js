/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import MainMenuSettingsControlsComponent from '../app/ui/component/MainMenuSettingsControls/MainMenuSettingsControlsComponent';

describe('MainMenuSettings', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <MainMenuSettingsControlsComponent 
                setActiveOption={ jest.fn() }
                type={ '' }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <MainMenuSettingsControlsComponent 
                setActiveOption={ jest.fn() }
                type={ '' }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
