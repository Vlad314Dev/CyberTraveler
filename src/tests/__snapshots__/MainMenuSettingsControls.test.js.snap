// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuSettings Should render layout 1`] = `
Array [
  <React.Fragment>
    <div
      className="SettingsControls__Wrapper"
    >
      <div
        className="SettingsControls__Image SettingsControls__Image_type_"
      />
    </div>
    <div
      className="MainMenuOption__List"
    >
      <div
        className="MainMenuOption__Wrapper"
      >
        <span
          className="MainMenuOption__Value"
          onClick={[Function]}
        >
          Back
        </span>
      </div>
    </div>
  </React.Fragment>,
]
`;
